Silicon/CommonSocPkg/Library/GpioLib/GpioInit.c

```diff
@@ -658,71 +662,96 @@ FillGpioTable (
   If the pins are not part of GPIO CFG DATA, call GpioPadConfigTable() directly
   with the appropriate arguments.
 
+  @param    Tag         Tag ID of the Gpio Cfg data item
+  @param    Entries     Number of entries in Gpio Table
+  @param    DataBuffer  Pointer to the Gpio Table to be programmed
+
   @retval EFI_SUCCESS                   The function completed successfully
   @retval EFI_NOT_FOUND                 If Gpio Config Data cant be found
 **/
 EFI_STATUS
 EFIAPI
 ConfigureGpio (
-  VOID
+  IN  UINT16  Tag,
+  IN  UINT32  Entries,
+  IN  UINT8   *DataBuffer
   )
 {
-  GPIO_CFG_HDR_INFO       GpioCfgCurrHdrInfo;
-  GPIO_CFG_HDR_INFO       GpioCfgBaseHdrInfo;
-  GPIO_CFG_HDR_INFO      *GpioCfgHdrInfo;
-  UINT32                  GpioEntries;
-  UINT32                  Index;
-  UINT32                  Offset;
-  UINT8                  *GpioCfgDataBuffer;
-  UINT8                  *GpioTable;
-  EFI_STATUS              Status;
+  ARRAY_CFG_HDR  *GpioCfgCurrHdr;
+  ARRAY_CFG_HDR  *GpioCfgBaseHdr;
+  ARRAY_CFG_HDR  *GpioCfgHdr;
+  UINT32         GpioEntries;
+  UINT32         Index;
+  UINT32         Offset;
+  UINT8          *GpioCfgDataBuffer;
+  UINT8          *GpioTable;
 
-  GpioEntries    = 0;
+  //
+  // Find either Tag to read CfgData or parse the GpioTable
+  //
+  if ((Tag == 0x00) && (Entries == 0 || DataBuffer == NULL) ) {
+    DEBUG ((DEBUG_INFO, "Provide either Tag or Gpio Table info!\n"));
+    return EFI_UNSUPPORTED;
+  }
 
   //
-  // Find the GPIO CFG HDR INFO
+  // If no Tag provided, check for GpioTable info;
+  // If    Tag provided, GpioTable params are don't care
   //
-  Status = GpioGetCfgHdrInfo (&GpioCfgCurrHdrInfo, 0xFF);
-  if (Status != EFI_SUCCESS) {
-    return Status;
+  if ((Tag == 0x00) && (Entries != 0 && DataBuffer != NULL) ) {

```

Comment:
Here only need check if Tag is 0 or not since Line 692 ~ 695 already checked invalid cases for Entries and DataBuffer. 
It would be nice if we could define a MICRO for the TAG 0. e.g.
#define    CDATA_NO_TAG     0